### PR TITLE
Cleaup readme

### PR DESCRIPTION
**Pull request by mleisa**
- **This repository allows merge commits, if you're creating a Release, make sure to select 'Squash and merge'**
- **When merging subtree updates, it is important to select 'Create a merge commit' when merging.**

This PR updates README.md with
- cleanup
- add upstream doc
- info about k8s ingress

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,12 +5,21 @@
...
 
 ## Introduction
-AWS Load Balancer controller manages the following AWS resources
+[AWS Load Balancer](https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html) controller manages the following AWS resources
 - Application Load Balancers to satisfy Kubernetes ingress objects
 - Network Load Balancers to satisfy Kubernetes service objects of type LoadBalancer with appropriate annotations
 
-## Security updates
-**Note**: Deployed chart does not receive security updates automatically. You need to manually upgrade to a newer chart.
+If your use case does not require an NLB, consider using the Kubernetes In-Tree [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) instead of the AWS Load Balancer Controller. An AWS Application Load Balancer can be provisioned when creating a Kubernetes `Ingress` in which case the AWS Load Balancer Controller may not be needed.
```
Comment (ubergesundheit) [inline README.md:12]: Maybe I'm wrong but I've never seen aws lb annotations on Ingress objects. Always Services (which are backing Ingress objects)

I think you should link to Giant Swarm docs. [This page](https://docs.giantswarm.io/advanced/ingress/service-type-loadbalancer/#customizing-the-external-load-balancer) partially handles AWS load balancers

Comment (ubergesundheit) [reply README.md:12]: Never used AWS ALBs though. Maybe it works with Ingresses?

File: README.md [hunk 2]
```diff
@@ -19,9 +28,12 @@ The controller runs on the worker nodes, so it needs access to the AWS ALB/NLB r
 IAM permissions can be setup through the kiam-app.
 
 Download the recommended IAM policy for the AWS Load Balancer Controller
-    ```
-    curl -o iam-policy.json https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/install/iam_policy.json
-    ```
+```bash
```
Comment (ubergesundheit) [inline README.md:31]: Can you check if this is still required? I think newer Giant Swarm platform releases have this stuff pre-configured.

To check: Create a test cluster with a new-ish platform release version on AWS, install this app, Add a Service with some AWS load-balancer-controller annotation, then check the logs of the aws load-balancer-controller pods to see if there are any errors.
If you want, we can pair on this

File: README.md [hunk 3]
```diff
@@ -19,9 +28,12 @@ The controller runs on the worker nodes, so it needs access to the AWS ALB/NLB r
...
+curl -o iam-policy.json https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/main/docs/install/iam_policy.json
+```
+
+## Security Updates
+**Note**: Deployed chart does not receive security updates automatically. You need to manually upgrade to a newer chart.
```
Comment (mcharriere) [inline README.md:36]: not sure what this means. I'd probably delete it (chart updates are a matter of updating the app, as the rest of our apps.)

File: README.md [hunk 4]
```diff
@@ -5,12 +5,21 @@
 AWS Load Balancer controller Helm chart for Giant Swarm clusters
 
 ## Introduction
-AWS Load Balancer controller manages the following AWS resources
+[AWS Load Balancer](https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html) controller manages the following AWS resources
```
Comment (mcharriere) [inline README.md:8]:
```suggestion
[AWS Load Balancer Controller](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/) manages the following AWS resources
```

File: README.md (hunk 1 above)
Comment (mcharriere) [reply README.md:12]: > As things are a bit trickier.

Let me elaborate:
- the `in-tree` and the `cloud-controller` implementations are feature equivalent. Both are frozen when it comes to Load balancer features.
- The `in-tree` and the cloud-controller` implementations support Classic LBs and partially NLBs. 
- Any new development/feature goes to the `aws-load-balancer-controller`
- The `aws-load-balancer-controller` supports NLBs (services) and ALBs (ingresses). NLB support here is more complete than the other 2 implementations.

Eventually we would need to make this app a default app for AWS, so I don't think it's worth to discourage its usage .

Comment (mleisa) [reply README.md:12]: I wasn't aware of the difference between `in-tree` and `cloud-controller`, thanks for that

Taking what @mcharriere said into consideration, it doesn't seem future-proof to promote the usage of just the kubernetes ingress over `aws-load-balancer-controller`, so I will remove that part

File: README.md (hunk 2 above)
Comment (mleisa) [reply README.md:31]: Are CAPA clusters already in usage? If yes, I'll add a note that it's required on CAPA

Comment (Gacko) [reply README.md:30]: I'm not sure if the role has been created by someone else manually and not been deleted or if it's really created automatically... would double check this.

File: README.md [hunk 5]
```diff
@@ -89,6 +98,12 @@ region: eu-west-1
 
 See our [full reference page on how to configure applications](https://docs.giantswarm.io/app-platform/app-configuration/) for more details.
 
+## Contributing & Reporting Bugs
+If you have suggestions for how `aws-load-balancer-controller` could be improved, or want to report a bug, open an issue! We'd love all and any contributions.
+
+Check out the [Contributing Guide](https://github.com/giantswarm/aws-load-balancer-controller-app/blob/main/CONTRIBUTING.md) for details on the contribution workflow, submitting patches, and reporting bugs.
+
+
```
Comment (Gacko) [inline README.md:105]:
```suggestion
```

**Review by Gacko — CHANGES_REQUESTED** [review]
Basically fine, but either we do not tell "it's not required for Vintage, only for CAPI" regarding the IAM stuff or we double check the current situation.

IIRC `kiam` is not in use any more for newer AWS releases, so probably and especially not for CAPI. This does not make a difference since you still need to create the role, it only changes the way this role is getting assumed.

Last but not least I'd rather be safe than sorry and ask the user to always ensure the role. Doesn't hurt...

File: README.md (hunk 2 above)
Comment (mcharriere) [reply README.md:30]: on vintage, the role is created by https://github.com/giantswarm/aws-operator/blob/master/service/controller/resource/tccpn/template/template_main_iam_policies.go
On CAPA the role is not created at all, thus giantswarm/giantswarm#26467

**Review by Gacko — APPROVED** [review]

**Pull request merged**